Public/Src/Engine/UnitTests/Scheduler/PipExecutorTest.cs

```diff
@@ -1033,6 +1033,94 @@ public async Task FailingProcessWithChunkSizeErrorAndErrorRegex()
         }
 
+        [Theory]
+        [InlineData(true, EventId.PipProcessError)]
+        [InlineData(false, EventId.PipProcessError)]
+        [InlineData(false, EventId.PipProcessError, 10 * SandboxedProcessPipExecutor.OutputChunkInLines)]
+        [InlineData(true, EventId.PipProcessWarning)]
+        [InlineData(false, EventId.PipProcessWarning)]
+        public async Task ProcessPrintPathsToLog(bool regexMatchesEverything, EventId eventId, int errorMessageLength = 0)
+        {
+            const string BadContents = "Anti-Matches!";
+
+            await WithCachingExecutionEnvironment(
+                GetFullPath(".cache"),
+                async env =>
+                { 
+                    string workingDirectory = GetFullPath("work");
+                    AbsolutePath workingDirectoryAbsolutePath = AbsolutePath.Create(env.Context.PathTable, workingDirectory);
+
+                    string destination = GetFullPath("dest");
+                    AbsolutePath destinationAbsolutePath = AbsolutePath.Create(env.Context.PathTable, destination);
+
+                    File.WriteAllText(destination, BadContents);
+                    if (eventId == EventId.PipProcessError)
+                    {
+                        Process pip = CreateErrorProcess(
+                            env.Context,
+                            workingDirectoryAbsolutePath,
+                            destinationAbsolutePath,
+                            errorPattern: regexMatchesEverything ? ".*" : "ERROR",
+                            errorMessageLength: errorMessageLength);
+                        var testRunChecker = new TestRunChecker();
+                        await testRunChecker.VerifyFailed(env, pip);
+                        AssertErrorEventLogged(eventId, errorMessageLength > 0 ? 3 : 1);
+                    }
+                    else
+                    {
+                        Process pip = CreateWarningProcess(
+                            env.Context, 
+                            workingDirectoryAbsolutePath, 
+                            destinationAbsolutePath,
+                            regexMatchesEverything ? false : true);
+                        var testRunChecker = new TestRunChecker();
+                        testRunChecker.ExpectWarning();
+                        await testRunChecker.VerifySucceeded(env, pip); 
+                        AssertWarningEventLogged(eventId);
+                    }
+
+                    string log = EventListener.GetLog();
+                    string relatedLog = GetRelatedLog(log, eventId);
+                    if (regexMatchesEverything)
+                    {
+                        XAssert.IsFalse(relatedLog.Contains(destination));
+                    }
+                    else
+                    {
+                        XAssert.IsTrue(relatedLog.Contains(destination));
+                    }

```

Comment:
I'd expect the condition to make your asserts wrong if the regex matches everything but the error message is still truncated due to the length being too long. Please add a test case for that.

Reply:
I think I covered this case with                         

> if (errorReader.Peek() == -1 && outReader.Peek() == -1)
> {
>         if (stdOutTotalLength != result.StandardOutput.Length || stdErrTotalLength != result.StandardError.Length)
>     {
>         errorWasTruncated = true;
>     }
>  }

Reply:
And I added this case         
`[InlineData(true, EventId.PipProcessError, 10 * SandboxedProcessPipExecutor.OutputChunkInLines)]`

Reply:
On further inspection I see on line 1092 that the test case changes the Pip's outputReportingMode dependeing on the errorMessageLength. So the scenario of the message getting truncated due to length still isn't exercised with that test case.

The test case I'm concerned about has the following conditions:
1. No regex filtering is applied
1. OutputReportingMode is set to TruncatedOutputOnError
1. The standard output stream exceeds the length at which it would be truncated

In that case the message will be truncated due to length, not regex matching.